dev/src/lobster/flat_gltf.h

```diff
@@ -0,0 +1,789 @@
+
+  return tex;
+}
+
+void SetTexture(Texture tex, GLuint texSlot) {
+  if (texSlot == 0)
+    drawTexture0 = tex;
+  if (tex.success) {
+    glActiveTexture(GL_TEXTURE0 + texSlot);
+    glBindTexture(GL_TEXTURE_2D, tex.texture);
+  }
+}
+
+} // namespace fgl
+
+namespace fgltf {
+
+bool verbose = true;
+
+#ifndef _VERBOUT_
+#define _VERBOUT_                                                              \
+  if (verbose)                                                                 \
+  std::cout
+#endif
+
+#ifndef _ERROUT_
+#define _ERROUT_ std::cerr
+#endif
+
+#ifndef MAX_SCENE_NODES
+#define MAX_SCENE_NODES 1000000
+#endif
+
+typedef struct {
+  int baseColorImage;
+  GLfloat metallicFactor;
+  GLfloat roughnessFactor;
+} SceneMaterial;
+
+SceneMaterial newSceneMaterial = {-1, 0.0f, 0.5f};
+
+typedef struct {
+  fgl::Mesh mesh;
+  int material;
+  int mode;
+} ScenePrim;
+
+typedef std::vector<ScenePrim> SceneMesh;
+
+typedef struct {
+  int mesh;
+  float3 translation;
+  float4 rotation;
+  float3 scale;
+  std::vector<int> children;
+} SceneNode;
+
+typedef struct {
+  int node;
+  int type;
+  int interpolation;
+  std::vector<float3> translations;
+  std::vector<float4> rotations;
+  std::vector<float3> scales;
+  std::vector<float> weights;
+} SceneAnimationData;
+
+typedef struct {
+  std::string name;
+  std::vector<SceneAnimationData> channels;
+} SceneAnimation;
+
+typedef struct {
+  std::vector<SceneMaterial> materials;
+  std::vector<SceneMesh> meshes;
+  std::vector<SceneNode> nodes;
+  std::vector<SceneAnimation> animations;
+  std::vector<int> roots;
+} Scene;
+
+Scene newScene = {{}, {}, {}, {}, {}};
+
+typedef struct {
+  std::string path;
+  fgl::Texture texture;
+} Image;
+
+std::vector<Image> images;
+
+int LoadImage(std::string gltfPath, std::string imgName, bool filter = true) {
+  // loads an image relative to the specified gltf path
+  // never loads the same image twice in the same process
+
+  // set the file path to the same folder as the glTF file
+  std::string imgFilePath =
+      gltfPath.substr(0, gltfPath.find_last_of('/') + 1) + imgName;
+  // try to find the image among already loaded images
+  for (size_t i = 0; i < images.size(); i++)
+    if (images[i].path == imgFilePath)
+      return i;
+  // no image found, so load the named file
+  int imgWidth, imgHeight, imgChannels;
+  std::string fileData;
+  if (LoadFile(imgFilePath.c_str(), &fileData) < 0) {
+    _ERROUT_ << "Failed to open " << imgFilePath << std::endl << std::endl;
+    return -1;
+  }
+  unsigned char *imgData =
+      stbi_load_from_memory((stbi_uc*)fileData.data(), fileData.size(), &imgWidth, &imgHeight, &imgChannels, 0);
+  fgl::Texture imgTexture =
+      fgl::LoadTexture(imgData, imgWidth, imgHeight, imgChannels, filter, filter);
+  stbi_image_free(imgData);
+  images.push_back({imgFilePath, imgTexture});
+  return images.size() - 1;
+}
+
+Scene LoadScene(std::string gltfPath, std::string schemaPath = "gltf.fbs") {
```

Comment:
Make the user specify where the schema is, i.e. all the way down to `gl_new_scene_gltf`

Reply:
As an optional parameter, perhaps? I was under the impression that the path to the `default.materials` file was hard-coded.

Reply:
Yup, that's because it is needed in every single lobster program. Giving the user some control over files is generally good. Wouldn't make it optional.

Reply:
`error: missing arg to builtin function: gltf_new_scene`

Huzzah, I just made it "better".